### PR TITLE
add the function of CI

### LINKED ISSUE #4: Implement the CI by Using Github Actions
## Task 
Imrement the CI and test them by using Github Actions

I added the ci.yaml file.

### PR DESCRIPTION
**Pull request by Rwatana**
Add the CI

### CODE REVIEW

**Comment by Rwatana** [comment]
@nayuta-ai 
## Topic
I think I have added an CI function by using Github actions and test the file.
Please check my code.

## Time
2h

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Go
+
+on:
+  push:
```
Comment (nayuta723) [inline .github/workflows/ci.yaml:4]: Please change the target on push event.
This implementation only supports this branch, but ci should support all pull request.

File: test/src/resource_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package main
```
Comment (nayuta723) [inline test/src/resource_test.go]: Divide into test file and src file.

Comment (nayuta723) [inline test/src/resource_test.go]: This file structure is strange.
test directory contains only test file.

Comment (nayuta723) [reply test/src/resource_test.go]: src directory contains some implementation, but containing only test file is strange.

**Review by nayuta723 — CHANGES_REQUESTED** [review]

**Comment by Rwatana** [comment]
@nayuta-ai 
## Topic
- I think I changed to work CI in all branches by changing them below. But, I couldn't
confirm them. Does it meet the demands?
```
on:
  push:
    branches: ['*']
  pull_request:
    branches: ['*']
```
- I changed the structure of directory. Please check my change.

File: test/src/resource_test.go (hunk 2 above)
Comment (nayuta723) [reply test/src/resource_test.go]: Refer to https://github.com/okuda-seminar/client-go/pull/2#discussion_r1311178569

**Comment by nayuta723** [comment]
@Rwatana 
Yes, It does!
Done!

**Comment by nayuta723** [comment]
Also, PR3 should only contain CI code. But it contains the test file related to PR2. Because it is strange, so please change.

**Pull request closed without merging**